src/usual/PLUsMgvStrat.sol

```diff
@@ -0,0 +1,124 @@
+  address public _usualDapp;
+  uint16 public _fee = 30; // Fee is in 10_000 = 100%
+  uint16 maxFee = 100;
+
+  constructor(
+    address admin,
+    IMangrove mgv,
+    UsualTokenInterface pLUsDAOToken,
+    MetaPLUsDAOToken metaPLUsDAOToken,
+    IERC20 usUSD
+  ) Direct(mgv, NO_ROUTER, 100_000) {
+    _pLUsDAOToken = pLUsDAOToken;
+    _metaPLUsDAOToken = metaPLUsDAOToken;
+    _pLUsDAOToken.approve(address(_metaPLUsDAOToken), type(uint).max);
+    _usUSD = usUSD;
+    if (admin != address(this)) {
+      setAdmin(admin);
+    }
+  }
+
+  modifier onlyDappOrAdmin() {
+    require(msg.sender == _usualDapp || msg.sender == admin(), "PLUsMgvStrat/onlyDappOrAdmin");
+    _;
+  }
+
+  function setFee(uint16 fee) public onlyAdmin {
+    require(fee <= maxFee, "PLUsMgvStrat/maxFee");
+    _fee = fee;
+    emit SetFee(fee);
+  }
+
+  function setUsualDapp(address usualDapp) public onlyAdmin {
+    _usualDapp = usualDapp;
+  }
+
+  function withdrawFees(address to) public onlyAdmin {
+    uint fee = _usUSD.balanceOf(address(this));
+    _usUSD.transfer(to, fee);
+    emit FeeWithdrawn(fee);
+  }
+
+  function __put__(uint amount, MgvLib.SingleOrder calldata order) internal virtual override returns (uint) {
+    uint fee = (amount * _fee) / 10_000;
+    _usUSD.transfer(offerIdToOwner[order.offerId], amount - fee);
+    emit FeePaid(fee);
+    return 0;
+  }
+
+  function __get__(uint amount, MgvLib.SingleOrder calldata order) internal virtual override returns (uint missing) {
+    _pLUsDAOToken.transferFrom(offerIdToOwner[order.offerId], address(this), amount);
+    return 0;
+  }
+
+  function newOffer(uint wants, uint gives, uint pivotId, address owner)
+    public
+    payable
+    onlyDappOrAdmin
+    returns (uint offerId)
+  {
+    offerId = _newOffer(
+      OfferArgs({
+        outbound_tkn: _metaPLUsDAOToken,
+        inbound_tkn: _usUSD,
+        wants: wants,
+        gives: gives,
+        gasreq: offerGasreq(),
+        gasprice: 0,
+        pivotId: pivotId,
+        fund: msg.value,
+        noRevert: false
+      })
+    );
+    offerIdToOwner[offerId] = owner;
+  }
+
+  function __posthookSuccess__(MgvLib.SingleOrder calldata order, bytes32 makerData)
+    internal
+    override
+    returns (bytes32)
+  {
+    return MangroveOffer.__posthookSuccess__(order, makerData);
+  }
+
+  function updateOffer(uint wants, uint gives, uint pivotId, uint offerId, address owner)
+    public
+    payable
+    onlyDappOrAdmin
+    returns (bytes32)
+  {
+    require(offerIdToOwner[offerId] == owner, "PLUsMgvStrat/wrongOwner");
+    return _updateOffer(
+      OfferArgs({
+        outbound_tkn: _metaPLUsDAOToken,
+        inbound_tkn: _usUSD,
+        wants: wants,
+        gives: gives,
+        gasreq: type(uint).max, // uses the old gasreg
+        gasprice: 0, // igonred
+        pivotId: pivotId,
+        noRevert: true,
+        fund: msg.value
+      }),
+      offerId
+    );
+  }
+}
```

Comment:
don't you need to override `__activate__` and `__checkList__` ? (i.e do `_pLUsDAOToken`
and `_metaPLUsDAOToken` etc. have the necessary approvals)?

Reply:
No i don't believe a have to here. Because Mangrove is only trying to transfer Meta token from the strat. It is then the Meta token contract itself that tries to transfer the PLUsDAOToken. And the approve function on the Meta token also approves the the PLUsDAOToken. This means when the Meta Token is trying to transfer from the strat to the taker, then the strat already has approve this by calling activate.

Reply:
Oh, and i do an approve in the constructor of PLUsMgvStrat, that makes i possible.